x/poa/types/genesis.go

```diff
@@ -74,3 +78,43 @@ func validateGenesisStateValidators(validators []Validator) (err error) {
 	}
 	return
 }
+
+// validateGenesisStatePrivilegeAssignments validates privilege assignments in genesis.
+//
+// Ensures that:
+// - The operator and privilege are non-empty strings.
+// - There are no duplicated assignments.
+func validateGenesisStatePrivilegeAssignments(assignments []ValidatorPrivilegeAssignment) error {
+	if len(assignments) == 0 {
+		return nil
+	}
+
+	seen := make(map[string]bool)
+
+	for _, assignment := range assignments {
+		operator := assignment.GetOperatorBech32()
+		privilege := assignment.GetPrivilege()
```

Comment:
Is there a point validating that the privilege string is a valid one, as in it contains e.g: "bridge", not something unsupported? Ultimately it's not a big deal, seeing it comes from the genesis file, you'd expect peer review to find issues early here.

Reply:
Well, actually x/poa doesn't care what the privilege is. It can be any string hence the validation is quite shallow.